src/ucp/wireup/wireup.c

```diff
@@ -153,10 +157,13 @@ ucs_status_t ucp_wireup_msg_progress(uct_pending_req_t *self)
         break;
     }
 
-out:
+out_free_req:
     ucs_free((void*)req->send.buffer);
     ucs_free(req);
-    return UCS_OK;
+out:
+    UCS_ASYNC_UNBLOCK(&ep->worker->async);
+    ucs_assert((status == UCS_OK) || (status == UCS_ERR_NO_RESOURCE));
```

Comment:
should be done by transports / pending_req_add

Reply:
removed